### PR TITLE
<Feature: Take Photo>

### PR DESCRIPTION
**Pull request by Vychron**

### CODE REVIEW

File: Assets/Scripts/Features/Gameplay/Take Photo/GeneratePhoto.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    void GenerateNewPhoto() {
+       Renderer rend = _picture.GetComponent<Renderer>();
+        rend.material.mainTexture = ToTexture2D(_renderTexture);
+    }
+
+    /// <summary>
+    /// Returns a Texture2D generated from RenderTexture "rTex".
+    /// </summary>
+    /// <param name="rTex"></param>
+    /// <returns></returns>
```
Comment (waffel183) [inline Assets/Scripts/Features/Gameplay/Take Photo/GeneratePhoto.cs]: Add small description of "rTex" parameter
Also add what it returns in the comment, not just <returns></returns>

**Review by waffel183 — CHANGES_REQUESTED** [review]
Good commits, but lacking a bit of documentation at 1 function. Fix this first before merging

**Comment by empireboy** [comment]
There needs to be a private access modifier in front of your methods, as shown in the code conventions under "Access Modifiers". This will make it easier to read if your method is private, public or protected.

Also im not sure if you need those lines:
`
using System.Collections;
`
`
using System.Collections.Generic;
`
If you dont need them you need to remove them. You can also find this in the code conventions under "Using".

Code conventions:
https://docs.google.com/document/d/1ndhz6gM4dyjpbsxsPXH3Zr36-6QOn1e2/edit#

@waffel183 You can take this into account for the next pull request.

File: Assets/Scripts/Features/Gameplay/Take Photo/CreatePhotoDoubleTap.cs [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
 /// A script to tell the game to take a picture when double tapping.
 /// </summary>
 public class CreatePhotoDoubleTap : MonoBehaviour
 {
 
-    void Update()
+    /// <summary>
+    /// Call once every frame update.
+    /// </summary>
+    private void Update()
```
Comment (waffel183) [inline Assets/Scripts/Features/Gameplay/Take Photo/CreatePhotoDoubleTap.cs]: Good that you use private functions, but do not create an XML comment when using a private function/variable

File: Assets/Scripts/Features/Gameplay/Take Photo/GeneratePhoto.cs [hunk 3]
```diff
@@ -10,28 +8,41 @@ public class GeneratePhoto : MonoBehaviour
...
+    }
+
+    /// <summary>
+    /// Returns a Texture2D generated from a given RenderTexture.
     /// </summary>
-    /// <param name="rTex"></param>
-    /// <returns></returns>
-    Texture2D ToTexture2D(RenderTexture rTex)
+    /// <param name="renderTexture"></param>
+    private Texture2D ToTexture2D(RenderTexture renderTexture)
```
Comment (waffel183) [inline Assets/Scripts/Features/Gameplay/Take Photo/GeneratePhoto.cs]: The parameter name is more descriptive, but add a description on the param like this: `<param name="renderTexture">renderTexture is used for ... </param>`
Also describe the return value: `<returns>modified texture</returns>`
[https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/returns](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/returns)

**Review by waffel183 — CHANGES_REQUESTED** [review]
Better use of private functions, but do not add xml comments to private functions/variables. If the function is too vague, use multiline comments to explain the code without using xml comments. If the private function obvious (private int addTwoNumbers(int a, int b)) you should not use multiline comments.
Also add a description of any parameters in a function, and a description of what a function returns (if it returns something)

File: Assets/Scripts/Features/Gameplay/Take Photo/CreatePhotoDoubleTap.cs (hunk 2 above)
Comment (waffel183) [reply Assets/Scripts/Features/Gameplay/Take Photo/CreatePhotoDoubleTap.cs]: Sorry, this is my fault. This is not a problem

**Review by waffel183 — COMMENTED** [review]
Ignore the feedback about the XML comments on private functions, this is your own choice. This is my wrong feedback that I apologize for.

**Review by waffel183 — APPROVED** [review]

**Pull request merged**